models/tasks.js

```diff
@@ -194,13 +195,29 @@ const fetchUserTasks = async (username, statuses = [], field, order) => {
 };
 
 const getNewTask = async (skill, level) => {
-  const task = await tasksModel
-    .where("category", "==", skill)
-    .where("level", ">=", level)
-    .where("level", "<=", level + 2)
-    .where("status", "==", TASK_STATUS.AVAILABLE)
-    .limit(1)
-    .get();
+  const availableTasks = await tasksModel.where("status", "==", TASK_STATUS.AVAILABLE).get();
+  const idArray = [];
+  let task;
+
+  if (!availableTasks.empty) {
+    availableTasks.forEach((item) => idArray.push(item.id));
+
+    if (!skill) {
+      task = await ItemModel.where("itemtype", "==", "TASK")
+        .where("itemid", "in", idArray)
+        .where("levelnumber", ">=", 2)
+        .limit(1)
+        .get();
+    } else {
```

Comment:
What is the max size of idArray
And when filtering based on  `in idArray`, I think `in` combines up to 10 equality, So if idArray size is more  than 10 then what will it happen?

Ref: https://firebase.google.com/docs/firestore/query-data/queries#in_not-in_and_array-contains-any